From 97a900a1ddb19d810b10903d6c1aacdb6107e60d Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Wed, 29 Sep 2021 19:44:32 +0800
Subject: [PATCH] arm64: mm: add the code that is missed when upgrade kernel

When booting up on nxp ls/lx series platforms, system hangs
at init process. This issue caused by merging upstream commit
5e10f9887ed8("arm64: mm: Fix TLBI vs ASID rollover"). Because
some code is missed when there is conflict with sdk patch.
Therefore, add the missing code manually.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/include/asm/tlbflush.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/include/asm/tlbflush.h b/arch/arm64/include/asm/tlbflush.h
index 1d0b015e70f9..397b055ba322 100644
--- a/arch/arm64/include/asm/tlbflush.h
+++ b/arch/arm64/include/asm/tlbflush.h
@@ -256,6 +256,7 @@ static inline void flush_tlb_mm(struct mm_struct *mm)
 		dsb(ish);
 		isb();
 	} else {
+		asid = __TLBI_VADDR(0, ASID(mm));
 		__tlbi(aside1is, asid);
 		__tlbi_user(aside1is, asid);
 		dsb(ish);
@@ -274,6 +275,7 @@ static inline void flush_tlb_page_nosync(struct vm_area_struct *vma,
 		dsb(ish);
 		isb();
 	} else {
+		addr = __TLBI_VADDR(uaddr, ASID(vma->vm_mm));
 		__tlbi(vale1is, addr);
 		__tlbi_user(vale1is, addr);
 	}
-- 
2.17.1

